docs/reference/fields.txt

```diff
@@ -1079,3 +1081,70 @@ you can include the short versions of the modules.
+
+Field Names with Periods (``.``) and Dollar Signs (``$``)
+=========================================================
+
+Mongo 5.0 `adds specialized support
+<https://docs.mongodb.com/manual/core/dot-dollar-considerations/#std-label-crud-concepts-dot-dollar-considerations>`_
+for field names that are prefixed with a dollar sign (``$``) or
+that contain periods (``.``).
+
+Fields prefixed with ``$`` or containing ``.`` are not directly accessible in standard queries.
+They may only be accessed with special database helper methods (new in MongoDB 5.0)
+including ``$getField``, ``$setField``, and ``$literal``.
+In normal usage scenarios, such fields behave as "hidden" fields on the document.
+In addition, they come with severe limitations, such as the inability
+to be indexed.
+
+Support in Mongoid
+------------------
+
+This section outlines the behavior of Mongoid and the Mongo Ruby Driver
+when working with fields whose names are prefixed with ``$`` or contain ``.``
+("specially-named fields"):
+
+* Mongoid's :ref:`field <field-types>` macro does not support defining specially-named fields.
+* Mongoid does not support getting or setting the values of specially-named fields.
+* Mongoid does not support queries or operations which reference specially-named fields.
+* If a document existing in the database contains specially-named fields,
+  Mongoid will load the document normally **except** for its specially-named fields,
+  which will not be retrieved.
+* Mongoid will not change or overwrite any specially-named field's value when saving documents.
+* If support is added to Mongoid or the Mongo Ruby Driver in the future,
+  following the MongoDB behavior it will be implemented with specialized driver
+  functions specifically for accessing and writing specially-named fields.
+
+Further Background
+------------------
+
+Since its inception, MongoDB has reserved the ``$`` and ``.`` chars
+for `specific meanings in query operations <https://docs.mongodb.com/manual/crud/>`_.
+Support for ``$`` and ``.`` in field names is intended to address a use-case
+in storing data from third-party sources (i.e. not native to MongoDB.)
+Given the pre-existing query  usage of ``$`` and ``.``,
+such field names require special handling.
+
+In both MongoDB and Mongoid, ``$`` is reserved for operators and will
+return a syntax error (or will use the operator, if it exists):
+
+.. code-block:: ruby
+
+  Person.where('$my_money' => 100)
+  #=> raises Mongo::Error::OperationFailure (unknown top level operator: $my_money)
+
+In both MongoDB and Mongoid, ``.`` is reserved for embedded field references, and will return
+an empty result:
+
+.. code-block:: ruby
+
+  Person.where('www.mongodb.com' => 'favorite_website').to_a
+  #=> []
+  # queries for `www: { mongodb: { com: 'favorite_website' } }`
+
+Lastly, please note that method names containing ``$`` or ``.``
+are `not permitted in Ruby language
+<https://docs.ruby-lang.org/en/3.0.0/doc/syntax/methods_rdoc.html>`_.
```

Comment:
This is also not true:

```
define_method('foo.bar') { 42 }
send('foo.bar')
# => 42
```

We actually had a ticket filed by someone a while ago who generated method names with spaces in them based on MongoDB field names.

Reply:
That is hideous. And no, that doesn't count as a standard method because I cannot define it with `def` and call it as `object.foo.bar`.

I pray to the good lord that you are are not considering supporting `.` / `$` in this manner.

Reply:
At this point I am not aware of any plans for dot or dollar changes in Mongoid. In any event I would prefer to not break compatibility, hence I think we are largely on the same page with respect to Mongoid's direction but again I am unaware of any specific changes that are under consideration for Mongoid right now.